### PR TITLE
LPD-19669 Clear mapping of DSM fields in cards visualization mode

### PR DESCRIPTION
**Pull request by chestofwonders**
Comes from: https://github.com/liferay-frontend/liferay-portal/pull/4042

Task: https://liferay.atlassian.net/browse/LPD-19669

**Scope:**

**Allow users to clear and change assigned fields to DSM views in Cards visualization mode**

LPD-10735 FF should be added to access Visualization modes

image

When there is no a field mapped, the "Not Assigned" message should appear:

image

Once a field is mapped, a three-dot button should appears instead of the '+' button:

image

That button opens a dropdown menu that allows user to unmap the field (so the "Not Assigned" label should appear again, and the button should revert to the previous '+') or change the field. In that case, the selection fields modal is open, allowing the user to choose a new field.

image

**Acceptance Criteria:**

- Users can clear the assignment of fields in cards individually.
- When a user clicks on the action, the field is unmapped immediately and persisted.

### CODE REVIEW

**Comment by dsanz** [comment]
I don't see any concern in this PR

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx [hunk 1]
```diff
@@ -272,22 +310,73 @@ function CardsSection({
...
 }
+
+interface IUpdateButtonProps {
+	field?: IField;
+	label: string;
+	onClearSelection: () => void;
+	openSelectFieldModal: () => void;
+}
+
+function UpdateButton({
```
Comment (markocikos) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx:340]: This is a stateless common component, identical to [the one in List.tsx](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/list/List.tsx#L332-L381). As such, we can extract it to a `components` subdir in the right scope, into something like `/js/fds_view/visualization_modes/components/FieldAssignmentControls.tsx`. After this, we can update both List and Cards.

**Review by markocikos — COMMENTED** [review]
@chestofwonders The code looks good, just an improvement suggestion inline.

**Comment by chestofwonders** [comment]
Naming refactor requests from [previous PR](https://github.com/liferay-frontend/liferay-portal/pull/4042): https://github.com/liferay-frontend/liferay-portal/pull/4042#discussion_r1514248155 https://github.com/liferay-frontend/liferay-portal/pull/4042#discussion_r1514248232

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx (hunk 1 above)
Comment (chestofwonders) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/cards/Cards.tsx]: Done!

**Review by markocikos — COMMENTED** [review]
@chestofwonders Code looks good! Just one naming nit inline.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/components/FieldAssignmentControls.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+import {IField} from '../../../utils/types';
+
+interface IUpdateButtonProps {
+	field?: IField;
+	label: string;
+	onClearSelection: () => void;
+	openSelectFieldModal: () => void;
+}
+
+function UpdateButton({
```
[the remark this answers is not in the document]
Comment (chestofwonders) [reply modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/fds_view/visualization_modes/components/FieldAssignmentControls.tsx]: Done!

**Review by markocikos — APPROVED** [review]
@chestofwonders LGTM! :sparkles:

**Comment by chestofwonders** [comment]
ci:forward

**Comment by chestofwonders** [comment]
ci:report

**Comment by chestofwonders** [comment]
ci:report:

**Comment by chestofwonders** [comment]
ci:report

**Comment by chestofwonders** [comment]
ci:report:2549004870

**Comment by chestofwonders** [comment]
ci:forward:force

**Review by markocikos — CHANGES_REQUESTED** [review]
@chestofwonders Please merge https://github.com/liferay-frontend/liferay-portal/pull/4055 with this PR. This PR will break tests that the other PR fixes.

**Comment by chestofwonders** [comment]
PR updated with test fix from https://github.com/liferay-frontend/liferay-portal/pull/4055. We can forward it @markocikos @juanjofgliferay

**Review by juanjofgliferay — APPROVED** [review]
LGTM :rocket:

**Comment by chestofwonders** [comment]
ci:report:2547633387

**Comment by chestofwonders** [comment]
ci:test:relevant

**Comment by chestofwonders** [comment]
ci:forward

**Pull request closed without merging**